hazelcast/src/test/java/com/hazelcast/internal/util/ParallelOperationInvokerTest.java

```diff
@@ -0,0 +1,279 @@
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ParallelOperationInvokerTest extends HazelcastTestSupport {
+
+    private final TestHazelcastFactory factory = new TestHazelcastFactory();
+
+    private Config config;
+    private HazelcastInstance instance1;
+    private HazelcastInstance instance2;
+    private HazelcastInstance instance3;
+
+    @Before
+    public void setup() {
+        config = smallInstanceConfig();
+        config.getJetConfig().setEnabled(false);
+        config.setProperty(INVOCATION_RETRY_PAUSE.getName(), "10");
+        instance1 = factory.newHazelcastInstance(config);
+        instance2 = factory.newHazelcastInstance(config);
+        instance3 = factory.newHazelcastInstance(config);
+    }
+
+    @After
+    public void teardown() {
+        factory.terminateAll();
+    }
+
+    @Test
+    public void testInvoke() {
+        Node node = getNode(instance1);
+        Collection<UUID> uuids = invokeOnStableClusterParallel(
+                node.getNodeEngine(),
+                NoOpOperation::new,
+                0
+        ).join();
+
+        Collection<UUID> expectedUuids = getExpectedUuids(instance1, instance2, instance3);
+        assertThat(uuids)
+                .containsExactlyInAnyOrderElementsOf(expectedUuids);
+    }
+
+    @Test
+    public void testInvoke_withSingleMember() {
+        instance2.shutdown();
+        instance3.shutdown();
+
+        assertClusterSizeEventually(1, instance1);
+
+        Node node = getNode(instance1);
+        Collection<UUID> uuids = invokeOnStableClusterParallel(
+                node.getNodeEngine(),
+                NoOpOperation::new,
+                0
+        ).join();
+
+        Collection<UUID> expectedUuids = getExpectedUuids(instance1);
+        assertThat(uuids)
+                .containsExactlyInAnyOrderElementsOf(expectedUuids);
+    }
+
+    @Test
+    public void testInvoke_withThrowingTerminalException() {
+        Node node = getNode(instance2);
+        assertThatThrownBy(() -> {
+            invokeOnStableClusterParallel(
+                    node.getNodeEngine(),
+                    () -> new ExceptionThrowingOperation(new RuntimeException("expected")),
+                    0
+            ).join();
+        }).isInstanceOf(CompletionException.class)
+                .hasCauseInstanceOf(RuntimeException.class)
+                .hasStackTraceContaining("expected");
+    }
+
+    @Test
+    public void testInvoke_withThrowingCompletionExceptionWithNullCause() {
+        Node node = getNode(instance2);
+        assertThatThrownBy(() -> {
+            invokeOnStableClusterParallel(
+                    node.getNodeEngine(),
+                    () -> new ExceptionThrowingOperation(new CompletionException(null)),
+                    0
+            ).join();
+        }).isInstanceOf(CompletionException.class)
+                .hasCause(null);
+    }
+
+    @Test
+    public void testInvoke_withThrowingIgnorableException() {
```

Comment:
Can you duplicate this test for all 3 types of ignorable exceptions in addition to `HazelcastInstanceNotActiveException`?